### PR TITLE
Add IK range check program and dump data file 

### PR DESCRIPTION
**Pull request by knorth55**
## IK-solvable range check 
- green : ik-solvable in either arm 
- yellow : ik-solvable only in right arm
- blue : ik-solvable only in left arm 
- red: ik-unsolvable 

vizualize
visualize-disable
visualize-larm-able
visualize-rarm-able

### CODE REVIEW

**Comment by knorth55** [comment]
#1105

**Comment by k-okada** [comment]
Nice
- Please put this test program in baxtereus
- you can use 'for' and 'case', 'cond'
- how did you obtaikn ldump?
- .ldump could be .l, because this is just a lisp program 



--
◉ Kei Okada

2016/03/09 10:44、Shingo Kitagawa <notifications@github.com> のメッセージ:

**Comment by wkentaro** [comment]
I renamed it #1126 .
by the way, are there way to check variables in current scope?
I'd like to use it in the interactive shell after loading the loadable structure.

**Comment by knorth55** [comment]
OK, i will try to change it for interactive usage

**Comment by wkentaro** [comment]
Ah, the question is to @k-okada, I mean usage of roseus interactive shell.

**Comment by wkentaro** [comment]
- how did you obtaikn ldump?

```
roslaunch jsk_2015_05_baxter_apc baxter.launch
roslaunch jsk_2015_05_baxter_apc setup_torso.launch

ssh doura
roslaunch jsk_2015_05_baxter_apc setup_head.launch
exit

roseus $(rospack find jsk_2015_05_baxter_apc)/euslisp/main.l
> apc-init
> send *ri* :recognize-bins
> (mapcar #'(lambda (x) (cons (car x) (send *ri* :bbox->cube (cdr x)))) (send (*ri* . _bin-boxes) :list))
> (dict *)
```

**Comment by wkentaro** [comment]
Please squash the commits.

**Comment by knorth55** [comment]
modified: move to reset-pose before every ik calculation

**Comment by knorth55** [comment]
@k-okada created PR in jsk_robot/baxtereus https://github.com/jsk-ros-pkg/jsk_robot/pull/566

File: jsk_2016_01_baxter_apc/euslisp/test/ik-check.l [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                (send larm-step-cube :set-color :blue)
+                (setq *larm-able* (append *larm-able* (list larm-step-cube))))
+              ((and (not larm-bool) rarm-bool)
+                (setq rarm-step-cube (make-cube step-dist step-dist step-dist :pos pos))
+                (send rarm-step-cube :set-color :yellow)
+                (setq *rarm-able* (append *rarm-able* (list rarm-step-cube))))
+              (t nil))
+              (setq step-cube (make-cube step-dist step-dist step-dist :pos pos))
+              (send step-cube :set-color :green)
+              (setq *able* (append *able* (list step-cube))))
```
Comment (wkentaro) [inline jsk_2016_01_baxter_apc/euslisp/test/ik-check.l]:
```
(load "package://jsk_2015_05_baxter_apc/euslisp/jsk_2015_05_baxter_apc/util.l")
(setq a)
(pushback 1 a)
a
> (1)
```

File: jsk_2016_01_baxter_apc/euslisp/test/ik-check.l [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  (objects (append (list *baxter*) *able*)))
+
+(defun visualize-larm-able ()
+  (objects (append (list *baxter*) *larm-able*)))
+
+(defun visualize-rarm-able ()
+  (objects (append (list *baxter*) *rarm-able*)))
+
+(defun visualize-disable ()
+  (objects (append (list *baxter*) *disable*)))
```
Comment (wkentaro) [inline jsk_2016_01_baxter_apc/euslisp/test/ik-check.l]:
```
(objects (list* *baxter* *disable*))
```

**Comment by wkentaro** [comment]
Is ik-range.l necessary?

**Comment by knorth55** [comment]
it's dump file so you can regenerate same result by running the program

but the running the program takes long time so that i added it to check the result easily

**Comment by wkentaro** [comment]
Do you mean the ik-range.l is the result of ik-check.l program?
Why there is no line for saving in ik-check.l?

**Comment by knorth55** [comment]
i will add it
you mean you need no dump file but saving dump file line ? or both?

**Comment by wkentaro** [comment]
I mean I don't understand why the dumped file is necessary. 
You can explain about it by adding a saving line or documentation.

**Comment by knorth55** [comment]
I removed dump file

**Comment by wkentaro** [comment]
LGTM

**Pull request merged**